modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SSource.java

```diff
@@ -324,6 +315,38 @@ void write(MethodWriter writer, Globals globals) {
             writer.visitInsn(Opcodes.ACONST_NULL);
             writer.returnValue();
         }
+
+        writer.mark(endTry);
```

Comment:
I'm definitely conflicted here; you make good arguments.  I would just leave this as you have it for now then and it's something we will have to address when we split off down the road.  Maybe users pick the exceptions that should be caught as part of a script, but that's a discussion for a different PR for sure.